utils_nlp/models/transformers/abstractive_summarization_bertsum.py

```diff
@@ -0,0 +1,832 @@
+            [
+                fit_to_block_size(story, block_size, self.tokenizer.pad_token_id)
+                for story, _ in encoded_text
+            ]
+        )
+        encoder_token_type_ids = compute_token_type_ids(
+            encoded_stories, self.tokenizer.cls_token_id
+        )
+        encoder_mask = build_mask(encoded_stories, self.tokenizer.pad_token_id)
+
+        if train_mode and "tgt" in data[0]:
+            encoded_summaries = torch.tensor(
+                [
+                    [self.tgt_bos]
+                    + fit_to_block_size(
+                        summary, block_size - 2, self.tokenizer.pad_token_id
+                    )
+                    + [self.tgt_eos]
+                    for _, summary in encoded_text
+                ]
+            )
+            summary_num_tokens = [
+                encoded_summary.ne(self.tokenizer.pad_token_id).sum()
+                for encoded_summary in encoded_summaries
+            ]
+
+            Batch = namedtuple(
+                "Batch",
+                [
+                    "src",
+                    "segs",
+                    "mask_src",
+                    "tgt",
+                    "tgt_num_tokens",
+                    "src_str",
+                    "tgt_str",
+                ],
+            )
+            batch = Batch(
+                src=encoded_stories.to(device),
+                segs=encoder_token_type_ids.to(device),
+                mask_src=encoder_mask.to(device),
+                tgt_num_tokens=torch.stack(summary_num_tokens).to(device),
+                tgt=encoded_summaries.to(device),
+                src_str=stories,
+                tgt_str=summaries,
+            )
+        else:
+            Batch = namedtuple("Batch", ["src", "segs", "mask_src"])
+            batch = Batch(
+                src=encoded_stories.to(device),
+                segs=encoder_token_type_ids.to(device),
+                mask_src=encoder_mask.to(device),
+            )
+
+        return batch
+
+    def preprocess(self, story_lines, summary_lines=None):
+        """preprocess multiple data points
+
+           Args:
+              story_lines (list of strings): List of sentences.
+              targets (list of strings, optional): List of sentences.
+                  Defaults to None, which means it doesn't include summary and is
+                  not training data.
+
+            Returns:
+                If summary_lines is None, return list of list of token ids. Otherwise,
+                return a tuple of (list of list of token ids, list of list of token ids).
+
+        """
+        story_lines_token_ids = []
+        for line in story_lines:
+            try:
+                if len(line) <= 0:
+                    continue
+                story_lines_token_ids.append(
+                    self.tokenizer.encode(line, max_length=self.max_src_len)
+                )
+            except:
+                print(line)
+                raise
```

Comment:
maybe raise ValueError("Invalid line: {}".format(line))

Reply:
will add that